File: docs/guides/scala/src/main/scala/setSettingsThenSaveObjects.scala
import scala.concurrent.Future
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Await
import scala.concurrent.duration.Duration

import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions
import algoliasearch.search.IndexSettings

val getAppIDFor: String => String = _ => {
  "" // Implement your own logic here
}
val getIndexingApiKeyFor: String => String = _ => {
  "" // Implement your own logic here
}

def setSettingsThenSaveObjects(): Future[Unit] = {
  val playlists: Seq[Map[String, Any]] = Seq() // Your records

  playlists.foreach { playlist =>
    // Fetch from your own data storage and with your own code
    // the associated application ID and API key for this user
    val appID = getAppIDFor(playlist("user").toString)
    val apiKey = getIndexingApiKeyFor(playlist("user").toString)

    val client = SearchClient(appID, apiKey)
    val settings = IndexSettings(
      attributesForFaceting = Some(Seq("filterOnly(user)"))
    )

    Await.result(
      client.setSettings(
        indexName = "<YOUR_INDEX_NAME>",
        indexSettings = settings
      ),
      Duration(5, "sec")
    )

    Await.result(
      client.saveObjects(
        indexName = "<YOUR_INDEX_NAME>",
        objects = playlists
      ),
      Duration(5, "sec")
    )
  }

  Future.unit
}

File: docs/guides/javascript/src/setSettingsThenSaveObjects.ts
import { algoliasearch } from 'algoliasearch';

import type { IndexSettings } from 'algoliasearch';

const playlists: Record<string, any>[] = [
  /* Your records */
];

const getAppIDFor = (_: string) => {
  return ''; // Implement your own logic here
};
const getIndexingApiKeyFor = (_: string) => {
  return ''; // Implement your own logic here
};

playlists.forEach(async (playlist) => {
  // Fetch from your own data storage and with your own code
  // the associated application ID and API key for this user
  const appID = getAppIDFor(playlist['user']);
  const apiKey = getIndexingApiKeyFor(playlist['user']);

  try {
    const client = algoliasearch(appID, apiKey);
    const settings: IndexSettings = {
      attributesForFaceting: ['filterOnly(user)'],
    };
    await client.setSettings({ indexName: 'indexName', indexSettings: settings });

    await client.saveObjects({ indexName: 'indexName', objects: playlists });
  } catch (e: any) {
    console.error(e);
  }
});
